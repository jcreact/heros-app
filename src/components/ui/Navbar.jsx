import React from 'react';
import { NavLink } from 'react-router-dom';
import {
    Typography,
    AppBar,
    Toolbar,
    Button,
    IconButton,
    makeStyles,
} from '@material-ui/core';
import { ExitToApp } from '@material-ui/icons';
import gray from '@material-ui/core/colors/grey';

const useStyles = makeStyles((theme) => ({
    offset: theme.mixins.toolbar,
    appMenu: {
        flexGrow: 1,
        marginLeft: theme.spacing(4),
        marginRight: theme.spacing(2),
        '& > *': {
            color: gray[500],
        },
    },
    active: {
        color: 'white',
    },
}));

export const Navbar = () => {
    const classes = useStyles();

    return (
        <>
            <AppBar>
                <Toolbar>
                    <Typography variant="h6">Asociaciones</Typography>

                    <div className={classes.appMenu}>
                        <Button
                            component={NavLink}
                            to="/marvel"
                            color="inherit"
                            activeClassName={classes.active}
                        >
                            Marvel
                        </Button>
                        <Button
                            component={NavLink}
                            to="/dc"
                            color="inherit"
                            activeClassName={classes.active}
                        >
                            DC
                        </Button>
                    </div>

                    <IconButton
                        component={NavLink}
                        to="/login"
                        color="inherit"
                        variant="contained"
                    >
                        <ExitToApp />
                    </IconButton>
                </Toolbar>
            </AppBar>
            <div className={classes.offset} />
        </>
    );
};
