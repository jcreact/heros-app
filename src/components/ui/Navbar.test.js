import { mount } from 'enzyme';
import { MemoryRouter, Router } from 'react-router-dom';

import { AuthContext } from '../../auth/AuthContext';
import { types } from '../../types/types';

// Componente a probar
import { Navbar } from './Navbar';

describe(`Pruebas en el componente 'Navbar.jsx'.`, () => {

    const historyMock = {
        push: jest.fn(),
        replace: jest.fn(),
        location: {},
        listen: jest.fn(),
        createHref: jest.fn()
    };

    const contextValue = {
        dispatch: jest.fn(),
        user: { logged: true, name: 'Carlos' }
    };

    const component = mount(
        <AuthContext.Provider value={contextValue}>
            <MemoryRouter>
                <Router history={historyMock}>
                    <Navbar />
                </Router>
            </MemoryRouter>
        </AuthContext.Provider>
    );

    afterEach(() => jest.clearAllMocks());

    test('Debe renderizar el componente correctamente.', () => {
        expect(component).toMatchSnapshot();
        expect(component.find('header').exists()).toBeTruthy();
        expect(component.find('h6.MuiTypography-root').at(0).text()).toBe('Asociaciones');
        expect(component.find('h6.MuiTypography-root').at(1).text()).toBe('Carlos');
        expect(component.find('a[role="button"]').at(0).text()).toContain('Marvel');
        expect(component.find('a[role="button"]').at(1).text()).toContain('DC');
        expect(component.find('a[role="button"]').at(2).text()).toContain('buscar');

    });

    test(`Debe llamar el logout y usar el history`, () => {
        component.find('button').prop('onClick')();
        expect(contextValue.dispatch).toHaveBeenCalled();
        expect(contextValue.dispatch).toHaveBeenCalledWith({ type: types.logout });
        expect(historyMock.replace).toHaveBeenCalledWith('/login');
    });
});
