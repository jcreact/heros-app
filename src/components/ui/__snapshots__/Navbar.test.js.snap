// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente 'Navbar.jsx'. Debe renderizar el componente correctamente. 1`] = `
Array [
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <h6
        className="MuiTypography-root MuiTypography-h6"
      >
        Asociaciones
      </h6>
      <div
        className="makeStyles-appMenu-2"
      >
        <a
          aria-current={null}
          aria-disabled={false}
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <span
            className="MuiButton-label"
          >
            Marvel
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </a>
        <a
          aria-current={null}
          aria-disabled={false}
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <span
            className="MuiButton-label"
          >
            DC
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </a>
        <a
          aria-current={null}
          aria-disabled={false}
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <span
            className="MuiButton-label"
          >
            buscar
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </a>
      </div>
      <h6
        className="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorInherit"
      >
        Carlos
      </h6>
      <button
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10.09 15.59L11.5 17l5-5-5-5-1.41 1.41L12.67 11H3v2h9.67l-2.58 2.59zM19 3H5c-1.11 0-2 .9-2 2v4h2V5h14v14H5v-4H3v4c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </header>,
  <div
    className="makeStyles-offset-1"
  />,
]
`;
